File: dart/question9.dart
void main() {
  String reverse(String s) => s.split("").reversed.join();

  print(reverse("巭閄槑爨齉"));

  print(palindrome(12341254));
}

bool palindrome(int x) {
  String y = x.toString();
  String z = y.split("").reversed.join();
  if (y == z) {
    return true;
  }
  return false;
}

File: src/question9.rs

pub fn palindrome(x: i32) -> bool {
    let y = x.to_string();
    let z:String = y.chars().rev().collect();
    z == y
}


pub fn check_if_palindrome(s: String) -> bool {
    let list: Vec<char> = s.chars().collect();
    let mut left = 0;
    let mut right = s.len() - 1;
    while left < right {
        if list[left] != list[right]{
            return false;
        }
    left += 1;
    right -= 1;
    }
    true
}

#[test]
fn example_1() {
    let s = "asdfdsty";
    assert_eq!(check_if_palindrome(s.to_string()),false);
}